src/inference/src/dev/core_impl.cpp

```diff
@@ -289,9 +291,11 @@ ov::Parsed ov::parseDeviceNameIntoConfig(const std::string& deviceName, const An
         }
     };
 
-    // clean-up auto-batch related properties
-    clean_batch_properties(updated_device_name, updated_config, ov::hint::allow_auto_batching);
-    clean_batch_properties(updated_device_name, updated_config, ov::auto_batch_timeout);
+    // clean-up auto-batch related properties when api is not called from get_supported_property
+    if (!keep_batch_property) {
+        clean_batch_properties(updated_device_name, updated_config, ov::hint::allow_auto_batching);
```

Comment:
maybe we need to fix `clean_batch_properties` to remove AB properties only for HW devices?

Reply:
clean_batch_properties will not remove AB for virtual plugin, but this is at the stage of compiling model to hardware (because hardware does not support AB), but here in this issue this api is called from get_supported_properties, so from core perspective, AB should be always reported as supported instead of being cleaned up.

Reply:
Agree. Maybe we need to call this flag as keep core properties? To be more generic

Reply:
updated